packages/component-base/src/slot-controller.d.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { ReactiveController } from 'lit';
+
+export class SlotController implements ReactiveController {
+  constructor(
+    host: HTMLElement,
+    slotName: string,
+    slotFactory: () => HTMLElement,
+    slotInitializer: (host: HTMLElement, node: HTMLElement) => void
```

Comment:
Should these be optional?
<img width="936" alt="Screenshot 2021-12-13 at 14 25 02" src="https://user-images.githubusercontent.com/1222264/145811992-495c3c91-4acc-4fd6-ac14-f16dfb3ef801.png">


```suggestion
    slotFactory?: () => HTMLElement,
    slotInitializer?: (host: HTMLElement, node: HTMLElement) => void
```

Reply:
Thanks 👍